contract/src/lib.rs

```diff
@@ -73,258 +85,320 @@ mod state {
     }
 }
 
-struct ForkState {
-    fork_chain: Vec<state::Header>,
-    heaviest_block: usize,
-    chainwork: usize,
-}
-
 // Define the contract structure
 #[near(contract_state)]
 pub struct Contract {
-    // block headers received from Bitcoin relay service
-    // block_headers: Vec<state::Header>,
-    // block with the highest chainWork, i.e., blockchain tip
-
-    // fork_id -> vector of block headers from this fork
-    // we will promote one of the forks if we have to do a chain reorg
-    // when one of the forks will eventually reach the highest possible weight
-    // fork chainwork of fork is bigger than chainwork of the main chain
-    forks: near_sdk::store::LookupMap<usize, Vec<state::Header>>,
-
-    // fork_id -> latest_header
-    latest_fork_block_hashes: near_sdk::store::LookupMap<usize, state::Header>,
+    // A pair of lookup maps that allows to find header by height and height by header
+    mainchain_height_to_header: near_sdk::store::LookupMap<u64, String>,
+    mainchain_header_to_height: near_sdk::store::LookupMap<String, u64>,
 
-    // mapping(bytes32 => HeaderInfo) public _headers;
-    // mapping(uint256 => bytes32) public _mainChain; // mapping of block heights to block hashes of the MAIN CHAIN
+    // Block with the highest chainWork, i.e., blockchain tip, you can find latest height inside of it
+    mainchain_tip_blockhash: String,
 
-    height_to_header: near_sdk::store::LookupMap<usize, String>,
+    // Mapping of block hashes to block headers (ALL ever submitted, i.e., incl. forks)
+    headers_pool: near_sdk::store::LookupMap<String, state::Header>,
 
-    // mapping of block hashes to block headers (ALL ever submitted, i.e., incl. forks)
-    headers: near_sdk::store::LookupMap<String, state::Header>,
-
-    current_fork_id: usize,
-    // We use latest block to help offchain relayer to recover the reading of blocks and
-    // to understand if we currently writing a fork or not.
-    // latest_block_info: String
-
-    // block with the highest chainWork, i.e., blockchain tip
-    heaviest_block: String,
-
-    // highest chainWork, i.e., accumulated PoW at current blockchain tip
-    high_score: [u8; 32]
+    // If we should run all the block checks or not
+    enable_check: bool,
 }
 
 // Define the default, which automatically initializes the contract
 impl Default for Contract {
     fn default() -> Self {
         Self {
-            height_to_header: near_sdk::store::LookupMap::new(b"a"),
-            headers: near_sdk::store::LookupMap::new(b"h"),
-            forks: near_sdk::store::LookupMap::new(b"f"),
-            latest_fork_block_hashes: near_sdk::store::LookupMap::new(b"l"),
-            current_fork_id: 0,
-            heaviest_block: String::new(),
-            high_score: [0; 32],
+            mainchain_height_to_header: near_sdk::store::LookupMap::new(
```

Comment:
Let's remove the Default crate implementation, and move the this code to the init method

Reply:
done